Wrapping/Generators/Python/PyBase/pyBase.i

```diff
@@ -322,18 +322,10 @@ str = str
 
     %pythonprepend itkObject::AddObserver %{
         import itk
-        if len(args) == 3 and not issubclass(args[2].__class__, itk.Command) and callable(args[2]):
-            args = list(args)
+        if not issubclass(cmd.__class__, itk.Command) and callable(cmd):
             pycommand = itk.PyCommand.New()
-            pycommand.SetCommandCallable( args[2] )
-            args[2] = pycommand
-            args = tuple(args)
-        elif len(args) == 2 and not issubclass(args[1].__class__, itk.Command) and callable(args[1]):
-            args = list(args)
-            pycommand = itk.PyCommand.New()
-            pycommand.SetCommandCallable( args[1] )
-            args[1] = pycommand
-            args = tuple(args)
+            pycommand.SetCommandCallable( cmd )
+            cmd = pycommand
```

Comment:
With this commit, the pythonprepend code is very much simplified, as the last parameter of `AddObserver` now has a name, "cmd". Rather than either `args[2]` or `args[3]`.

Reply:
Does this function cause any problems: `Object::AddObserver(const EventObject & event, std::function<void(const EventObject &)> function) const`?

Reply:
@blowekamp It looks like the `AddObserver(const EventObject & event, std::function<void(const EventObject &)> function)` overload does not cause any problem, when looking at the CI. Do you think it _should_ cause problems?

Reply:
Is the Python wrapping supposed to support passing an `std::function<void(const EventObject &)>` as argument? Or is the `std::function<void(const EventObject &)>` support specific to the C++ interface?

_If_ the Python wrapping is supposed to support passing an `std::function<void(const EventObject &)>` as argument, _in Python_, does this pull request affect such support one way or another? Honestly that isn't clear to me. I just look at the CI, and the PR appears to pass the tests just fine.